src/context.hpp

```diff
@@ -49,23 +49,21 @@ struct cvk_context : public _cl_context,
         }
         // Get printf buffer size from extension.
         auto buff_size_prop_index =
-            get_property_index(CL_PRINTF_BUFFERSIZE_ARM);
+            get_property_value_index(CL_PRINTF_BUFFERSIZE_ARM);
         if (buff_size_prop_index != -1 && !config.printf_buffer_size.set) {
             m_printf_buffersize = m_properties[buff_size_prop_index];
         } else {
-            m_printf_buffersize = config.printf_buffer_size;
+            m_printf_buffersize = 0;
```

Comment:
I think it `m_printf_buffersize` really needs to be initialize with `config.printf_buffer_size()`.

Reply:
I changed the getter so the buffer size is taken from the config if it wasn't provided via the extension. This works for test overrides. It adds a bit of complexity but I think it's worth it for the benefits on the test side.